### ISSUE TITLE
Add error boundaries to the website

### ISSUE THREAD

**Issue by bvaughn** [reactjs/reactjs.org#9]
A few places where it could be handy:
* Live editor on the homepage to show crashes (or for issues like #3)
* Search box that gets broken by Fair Adblocker because a global is missing

**Comment by bvaughn** [comment]
This issue can also potentially help with #14

**Comment by ishankbahl** [comment]
Hey @bvaughn , wanted to work on this. Looks like issue #3 , #14 , https://github.com/facebook/react/issues/11008 can be solved by applying error boundaries around CodEditor, Header and Layout component where we just nudge people to whitelist the site in adblockers (As these are happening due to Fair Adblocker) . What are your thoughts?

**Comment by bvaughn** [comment]
Agreed that error boundaries could probably help with a couple of issues! Please feel free to take this! 😄

I've added an "in-progress" label so that others will know not to start work on the issue. If you change your mind about the issue, no worries! Just let me know so that I can remove the label and free it up for someone else to claim.

Cheers!

**Comment by bvaughn** [comment]
It looks like the remaining TODO here (the search box) isn't being actively worked on so I'm removing the in-progress label.

**Comment by SadPandaBear** [comment]
Hey @bvaughn I think #186 is fixing this now.

**Comment by bvaughn** [comment]
We've added specific error-handling for failed docsearch and Babel loads, but this issue still remains open to add at least one general purpose error boundary around the site in case of other, unexpected errors.

**Comment by M-ZubairAhmed** [comment]
Hi @bvaughn I see that error boundaries around components for handling add block and other issues are added.
I would like to understand 

> general purpose error boundary around the site.
do you mean would you like to place an error boundary around the main app component or around each single component?

**Comment by naveen1249** [comment]
..

**Comment by spiritanand** [comment]
Is this issue open? Can you assign it to me if it is?

**Comment by Rorqvists** [comment]
[``]($ cloudcannon files pull
Updating 273 files... done
$ cloudcannon run build
Preparing HTML files... done
$ source .cloudcannon/preinstall
Not found.
$ cd /usr/local/__site/src/
$ npm install
npm warn deprecated @codemirror/stream-parser@0.18.2: As of 0.20.0, this package has been merged into @codemirror/language
npm warn deprecated @codemirror/rectangular-selection@0.18.1: As of 0.20.0, this package has been merged into @codemirror/view
npm warn deprecated lezer-tree@0.13.2: This package has been replaced by @lezer/common
npm warn deprecated @codemirror/tooltip@0.18.4: As of 0.20.0, this package has been merged into @codemirror/view
npm warn deprecated @codemirror/basic-setup@0.18.2: In version 6.0, this package has been renamed to just 'codemirror'
npm warn deprecated @codemirror/panel@0.18.2: As of 0.20.0, this package has been merged into @codemirror/view
npm warn deprecated @codemirror/text@0.18.1: As of 0.20.0, this package has been merged into @codemirror/state
npm warn deprecated @codemirror/comment@0.18.1: As of 0.20.0, this package has been merged into @codemirror/commands
npm warn deprecated @codemirror/fold@0.18.2: As of 0.20.0, this package has been merged into @codemirror/language
npm warn deprecated @codemirror/matchbrackets@0.18.0: As of 0.20.0, this package has been merged into @codemirror/language
npm warn deprecated @codemirror/history@0.18.1: As of 0.20.0, this package has been merged into @codemirror/commands
npm warn deprecated @codemirror/gutter@0.18.4: As of 0.20.0, this package has been merged into @codemirror/view
npm warn deprecated @codemirror/rangeset@0.18.5: As of 0.20.0, this package has been merged into @codemirror/state
npm warn deprecated @codemirror/closebrackets@0.18.0: As of 0.20.0, this package has been merged into @codemirror/autocomplete
npm warn deprecated @codemirror/highlight@0.18.4: As of 0.20.0, this package has been split between @lezer/highlight and @codemirror/language
npm warn deprecated lezer@0.13.5: This package has been replaced by @lezer/lr

added 550 packages, and audited 551 packages in 27s

183 packages are looking for funding
  run `npm fund` for details

31 vulnerabilities (3 low, 17 moderate, 11 high)

To address issues that do not require attention, run:
  npm audit fix

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
npm notice
npm notice New major version of npm available! 10.7.0 -> 11.3.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v11.3.0
npm notice To update run: npm install -g npm@11.3.0
npm notice
$ cd /usr/local/__site/src/
$ source .cloudcannon/prebuild

> @example/basics@0.0.1 prebuild:fetch-theme-color
> node fetch-theme-color.cjs

$ cd /usr/local/__site/src/
$ npm run build

> @example/basics@0.0.1 build
> astro build

04:36:18 AM [content] Types generated 1.54s
04:36:18 AM [build] output target: static
04:36:18 AM [build] Collecting build info...
04:36:18 AM [build] Completed in 1.57s.
04:36:18 AM [build] Building static entrypoints...
04:36:30 AM [build] Completed in 11.60s.

 building client url)

**Comment by iamdivyanshgautam** [comment]
Hi @bvaughn, I'd like to work on this issue if it's still available. I've already forked the repo, set up the project locally, and started implementing the ErrorBoundary. Let me know if I can proceed — I'd be happy to open a PR soon.

**Comment by rickhanlonii** [comment]
This issue is no longer relevant on the new react.dev site I believe because error boundaries and tracking are built in I believe.
